tsl/src/data_node.c

```diff
@@ -638,6 +638,8 @@ data_node_add_internal(PG_FUNCTION_ARGS, bool set_distid)
 	bool PG_USED_FOR_ASSERTS_ONLY result;
 	DbInfo database;
 
+	PreventCommandIfReadOnly("add_data_node()");
```

Comment:
Isn't it better to put these blocks for TSL in the outer-wrapper functions, (i.e., in `cross_module_fn.c`)? That would make it easier to verify that all mutation functions have this block. We could also move the read_only test to non-TSL in that case.

Reply:
I had this idea, but thought it would be odd that apache part would do those checks explicitly. Plus I did not like the idea to separate this logic from actual implementation, because there could be more of them, like preventing running inside transaction block, etc. Otherwise we should separate all of our API functions into separate file and to those checks consistently there

Reply:
Ok, makes sense. I am on the fence myself, but figured it was good to raise the idea to get some pros and cons.

Happy to leave as is.